tensorflow_recommenders_addons/dynamic_embedding/core/kernels/redis_impl/redis_connection_util.hpp

```diff
@@ -0,0 +1,408 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#pragma once
+#include <aio.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <x86intrin.h>
+
+#include <cmath>
+#include <iostream>
+
+extern "C" {
+#include <hiredis/sds.h>
+}
+#include "tensorflow/core/framework/types.h"
+#include <sw/redis++/connection.h>
+#include <sw/redis++/connection_pool.h>
+#include <sw/redis++/redis++.h>
+#include <sw/redis++/sentinel.h>
+
+#include "md5.h"
+
+namespace tensorflow {
+namespace recommenders_addons {
+namespace redis_connection {
+/*
+if v=0 or 1, return itself. if v<0, v=abs(v).
+*/
+inline unsigned round_next_power_two_bitlen(int v) {
+  v = abs(v);
+  if ((v | 1) == 1)
+    return v;
+  // prevent v is already a power_two
+  --v;
+  // How many bits for storing v at least.
+  int bitlen = 1 + static_cast<int>(std::log2(v));
+  // v = 1 << bitlen; //number_round_next_power_two
+  return bitlen;
+}
+
+inline unsigned long get_file_size(const std::string path) {
+  unsigned long filesize = -1;
+  struct stat statbuff;
+  if (stat(path.data(), &statbuff) < 0) {
+    throw("The file " + path + " does not exist");
+    return filesize;
+  } else {
+    filesize = statbuff.st_size;
+  }
+  return filesize;
+}
+
+inline int createDirectory(const std::string path) {
+  int len = path.length();
+  char tmpDirPath[1024] = {0};
+  for (int i = 0; i < len; i++) {
+    tmpDirPath[i] = path[i];
+    if (tmpDirPath[i] == '\\' || tmpDirPath[i] == '/') {
+      if (access(tmpDirPath, 0) == -1) {
+        int ret = mkdir(tmpDirPath, S_IRWXU | S_IRWXG | S_IRWXO);
+        if (ret == -1)
+          return ret;
+      }
+    }
+  }
+  return 0;
+}
+
+inline std::string check_dir(const std::string path_in) {
+  std::string path(path_in);
+  if (path.back() != '/') {
+    path.push_back('/');
+  }
+  if (access(path.c_str(), 0) == -1) // if folder doesn't exist
+  {
+    std::cout << "folder " << path << " doesn't exist" << std::endl;
+    if (createDirectory(path) == 0) {
+      std::cout << "folder " << path << " was created" << std::endl;
+    } else {
+      std::cout << "folder " << path << " failed to create" << std::endl;
+    }
+  }
+  return path;
+}
+
+inline int ReadInt32FromEnvVar(const std::string &env_var_name,
```

Comment:
#include "tensorflow/core/util/env_var.h"，Is it better to use ReadInt64FromEnvVar?